sdk/translation/azure-ai-translation-text/azure/ai/translation/text/_patch.py

```diff
@@ -0,0 +1,137 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+
+from typing import Union
+from azure.core.pipeline import PipelineRequest
+from azure.core.pipeline.policies import ( SansIOHTTPPolicy, BearerTokenCredentialPolicy, AzureKeyCredentialPolicy )
+from azure.core.credentials import ( TokenCredential, AzureKeyCredential )
+
+from ._client import TextTranslationClient as ServiceClientGenerated
+
+DEFAULT_TOKEN_SCOPE = "https://api.microsofttranslator.com/"
+
+def patch_sdk():
+    """Do not remove from this file.
+
+    `patch_sdk` is a last resort escape hatch that allows you to do customizations
+    you can't accomplish using the techniques described in
+    https://aka.ms/azsdk/python/dpcodegen/python/customize
+    """
+
+class TranslatorCredential:
+    """ Credential for Translator Service. It is using combination of Resource key and region.
+    """
+    def __init__(self, key: str, region: str) -> None:
+        self.key = key
+        self.region = region
+
+    def update(self, key: str) -> None:
+        """Update the key.
+        This can be used when you've regenerated your service key and want
+        to update long-lived clients.
+        :param str key: The key used to authenticate to an Azure service
+        :raises: ValueError or TypeError
+        """
+        if not key:
+            raise ValueError("The key used for updating can not be None or empty")
+        if not isinstance(key, str):
+            raise TypeError("The key used for updating must be a string.")
+        self._key = key
+
+class TranslatorAuthenticationPolicy(SansIOHTTPPolicy):
+    """ Translator Authentication Policy. Adds both authentication headers that are required.
+    Ocp-Apim-Subscription-Region header contains region of the Translator resource.
+    Ocp-Apim-Subscription-Key header contains API key of the Translator resource.
+    """
+    def __init__(self, credential: TranslatorCredential):
+        self.credential = credential
+
+    def on_request(self, request: PipelineRequest) -> None:
+        request.http_request.headers["Ocp-Apim-Subscription-Key"] = self.credential.key
+        request.http_request.headers["Ocp-Apim-Subscription-Region"] = self.credential.region
+
+def get_translation_endpoint(endpoint, api_version):
+    if not endpoint:
+            endpoint = "https://api.cognitive.microsofttranslator.com"
+
+    translator_endpoint: str = ""
+    if "cognitiveservices" in endpoint:
+        translator_endpoint = endpoint + "/translator/text/v" + api_version
+    else:
+        translator_endpoint = endpoint
```

Comment:
Just checking - the "https://api.cognitive.microsofttranslator.com" endpoint doesn't need the api version tagged on it?

Reply:
Correct, it is required as `api-version` query parameter.